Added migration for Self-Serve Integration permissions

naz opened the pull request:
⚠️  this branch is a rebase over https://github.com/TryGhost/Ghost/pull/16461 cleanup the commits before merging.

refs https://github.com/TryGhost/Team/issues/2790
needs https://github.com/TryGhost/Ghost/pull/16461

This migration adds permissions for Self-Serve Migration Integration to have access to Admin APIs:

POST /ghost/api/admin/db
POST /ghost/api/admin/db/media/inline
POST /ghost/api/admin/members/upload

🚨 Note on the order of the migration. I've copied [the order of explore-related migrations](https://github.com/TryGhost/Ghost/tree/3623a652ba7b91d59c6f1183cf91a667357178a9/ghost/core/core/server/data/migrations/versions/5.3) as they were doing very similar thing. 
<img width="568" alt="image" src="https://user-images.githubusercontent.com/675397/227306562-69992e7a-e4f5-44bf-8515-087637a4383d.png">

naz committed:
Added migration for Self-Serve Integration permissions

refs https://github.com/TryGhost/Team/issues/2790

- This migration adds permissions for Self-Serve Migration Integration to have access to Admin APIs:

POST /ghost/api/admin/db
POST /ghost/api/admin/db/media/inline
POST /ghost/api/admin/members/upload
GET  /ghost/api/admin/tags/:id
GET  /ghost/api/admin/tags/slug/:slug

kevinansfield commented on ghost/core/test/integration/migrations/migration.test.js:
```diff
@@ -155,7 +155,6 @@ describe('Database Migration (special functions)', function () {
 
             permissions.should.havePermission('Browse Products', ['Administrator', 'Editor', 'Author', 'Admin Integration']);
             permissions.should.havePermission('Read Products', ['Administrator', 'Editor', 'Author', 'Admin Integration']);
-            permissions.should.havePermission('Edit Products', ['Administrator', 'Admin Integration']);
```
Was this deletion intentional?

naz replied:
Nope  👀

naz committed:
Added e2e test checking integration access

refs https://github.com/TryGhost/Team/issues/2790

- The Self-Serve Integration should only be accessible to the Owner and Admin user roles otherwise we risk accidental indirect increase in role permissions - Self-Serve Integration has permissions which editors/contributors don't have.
